### ISSUE TITLE
npm and github versions are way out of sync

### ISSUE THREAD

**Issue by jasonkarns** [howardabrams/node-mocks-http#105]
npm shows current version is 1.5.3 but github shows only 1.4.3

How is github so far behind? Are tags not getting pushed? Commits missing on github?

**Comment by howardabrams** [comment]
Doh!

I'll get better at keeping Github in sync.

By the way, if any one is interested in helping me maintain this project, I'd appreciate it.

**Comment by jasonkarns** [comment]
FYI, you can just do `npm version <patch|minor|major>` to bump versions when doing a release. That will bump the version in package.json, do a git commit and git tag automatically.

And if you add the following as a script to package.json:

```
"scripts": {
  "postversion": "npm publish && git push --follow-tags"
}
```

then it will automatically be published to npm and pushed to github all automatically. (Since it appears that even with 1.5.4 released on github, that all the tags between 1.4.3 and 1.5.4 are still missing.)

**Comment by scottnath** [comment]
Over at [Punchcard CMS](https://github.com/punchcard-cms) we've set up [our own repo](https://github.com/punchcard-cms/semantic-release) which uses [semantic release](https://github.com/semantic-release/semantic-release). Semantic release controls the semantic versions according to the emoji used in commit messages. ([our emoji cheatsheet](https://github.com/punchcard-cms/punchcard/blob/master/CONTRIBUTING.md#emoji-cheatsheet)). Then on merge of PRs it will version the release and publish to npm.

We have a [wiki page which includes instructions on setting this up for new Punchcard repos](https://github.com/punchcard-cms/punchcard/wiki/Setting-Up-a-New-Punchcard-Repository#semantic-release)
